ui/app/helpers/constants/routes.js

```diff
@@ -55,6 +55,64 @@ const SIGNATURE_REQUEST_PATH = '/signature-request'
 const DECRYPT_MESSAGE_REQUEST_PATH = '/decrypt-message-request'
 const ENCRYPTION_PUBLIC_KEY_REQUEST_PATH = '/encryption-public-key-request'
 
+// Used to pull a convenient name for analytics tracking events. The key must
+// be react-router ready path, and can include params such as :id for popup windows
+const PATH_NAME_MAP = {
+  [DEFAULT_ROUTE]: 'Home',
+  [UNLOCK_ROUTE]: 'Unlock Page',
+  [LOCK_ROUTE]: 'Lock Page',
+  [`${ASSET_ROUTE}/:asset`]: `Asset Page`,
+  [SETTINGS_ROUTE]: 'Settings Page',
+  [GENERAL_ROUTE]: 'General Settings Page',
+  [CONNECTIONS_ROUTE]: 'Connections Settings Page',
+  [ADVANCED_ROUTE]: 'Advanced Settings Page',
+  [SECURITY_ROUTE]: 'Security Settings Page',
+  [ABOUT_US_ROUTE]: 'About Us Page',
+  [ALERTS_ROUTE]: 'Alerts Settings Page',
+  [NETWORKS_ROUTE]: 'Network Settings Page',
+  [CONTACT_LIST_ROUTE]: 'Contact List Settings Page',
+  [`${CONTACT_EDIT_ROUTE}/:address`]: 'Edit Contact Settings Page',
+  [CONTACT_ADD_ROUTE]: 'Add Contact Settings Page',
+  [`${CONTACT_VIEW_ROUTE}/:address`]: 'View Contact Settings Page',
+  [CONTACT_MY_ACCOUNTS_ROUTE]: 'My Accounts List Settings Page',
+  [`${CONTACT_MY_ACCOUNTS_VIEW_ROUTE}/:account`]: 'View Account Settings Page',
+  [`${CONTACT_MY_ACCOUNTS_EDIT_ROUTE}/:account`]: 'Edit Account Settings Page',
+  [REVEAL_SEED_ROUTE]: 'Reveal Seed Page',
+  [MOBILE_SYNC_ROUTE]: 'Sync With Mobile Page',
+  [RESTORE_VAULT_ROUTE]: 'Restore Vault Page',
+  [ADD_TOKEN_ROUTE]: 'Add Token Page',
+  [CONFIRM_ADD_TOKEN_ROUTE]: 'Confirm Add Token Page',
+  [CONFIRM_ADD_SUGGESTED_TOKEN_ROUTE]: 'Confirm Add Suggested Token Page',
+  [NEW_ACCOUNT_ROUTE]: 'New Account Page',
+  [IMPORT_ACCOUNT_ROUTE]: 'Import Account Page',
+  [CONNECT_HARDWARE_ROUTE]: 'Connect Hardware Wallet Page',
+  [SEND_ROUTE]: 'Send Page',
+  [`${CONNECT_ROUTE}/:id`]: 'Connect To Site Confirmation Page',
+  [`${CONNECT_ROUTE}/:id${CONNECT_CONFIRM_PERMISSIONS_ROUTE}`]: 'Grant Connected Site Permissions Confirmation Page',
+  [CONNECTED_ROUTE]: 'Sites Connected To This Account Page',
+  [CONNECTED_ACCOUNTS_ROUTE]: 'Accounts Connected To This Site Page',
+  [`${CONFIRM_TRANSACTION_ROUTE}/:id${CONFIRM_TOKEN_METHOD_PATH}`]: 'Confirm Token Method Transaction Page',
```

Comment:
I think we should have `${CONFIRM_TRANSACTION_ROUTE}/:id` here as a route as well? :thinking: 

It would immediately redirect to one of these other routes, but it's still on that page until the redirect happens, during the render.

Reply:
I can add it, but we'd want to skip that page track unless it has some importance to the user flow? I have handled this route specifically in the page tracking code to skip tracking the root page before the redirect.

Reply:
Oh I see, I only just noticed this. This seems fine, nevermind!